src/moon.c

```diff
@@ -0,0 +1,79 @@
+/*
+
+ The MIT License (MIT)
+
+ Copyright (c) 2023 qlpqlp (twitter.com/inevitable360)
+ Copyright (c) 2022 The Dogecoin Foundation
+
+ Permission is hereby granted, free of charge, to any person obtaining
+ a copy of this software and associated documentation files (the "Software"),
+ to deal in the Software without restriction, including without limitation
+ the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ and/or sell copies of the Software, and to permit persons to whom the
+ Software is furnished to do so, subject to the following conditions:
+ The above copyright notice and this permission notice shall be included
+ in all copies or substantial portions of the Software.
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+ OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ OTHER DEALINGS IN THE SOFTWARE.
+
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+/**
+ * @brief Calling the moon, it returns the current moon phase
+ * @return moon icon
+ */
+
+char* moon()
+{
+  int D,c,e;
+  double jd;                      // Julian days
+
+  time_t t = time(NULL);          // get time
+  struct tm tm = *localtime(&t);  // get localtime
+  int y = tm.tm_year + 1900;      // year, because its only above 1900, we add 1900 to the year
+  int m = tm.tm_mon;              // month
+  int d = tm.tm_mday - 2;         // day, because it can show the moon with 2 days in advance, we subtract
+
+  if (m < 3) {
+    y--;
+    m += 12;
+  }
+  ++m;
+
+  c = 365.25*y;                   // 365.25 total days in a year
+  e = 30.6*m;
+  jd = c+e+d-694039.09;           // Julian days
+  jd /= 29.53;                    // moons cycle 29.53 days
+  D = jd;	                        // convert decimal to int 	   
+  jd -= D;                        // get negative of Julian days
+  D = jd*8 + 0.5;	                // get fraction from 0-8 by rounding 0.5 to it               
+
+  switch (D & 7) {                // because the new moon = 0 is the same on 8 we convert to 0
+    case 0:
+      return "🌚";               // New Moon Hype / Discounts
+    case 1:
+      return "🐶";               // Show Doge wen no moon
+    case 2:
+      return "🌓";               // First Quarter Moon
+    case 3:
+      return "🐶";               // Show Doge wen no moon
+    case 4:
+      return "🌝";               // Full Moon DownTrend / No Discounts
+    case 5:
+      return "🐶";               // Show Doge wen no moon
+    case 6:
+      return "🌗";               // Third Quarter
+    case 7:
+      return "🐶";               // Show Doge wen no moon
+  }
+  return 0;
```

Comment:
Default will return 0 (failure) so it's necessary for a new moon to be 1 and so on.

Reply:
dono how C handles `return`, but because it always return a `char*`, the moon does it call also `return 0` ?

Reply:
Paulo this looks great but I guess a few things: 

1. I think we should try returning a char instead of a char*.  On systems that support unicode display, char alone (vs char*) should be able to store the symbol and return it.  It also makes more sense in useage as a fully terminated string carries excess and can mess with UI formatting..

2. I'm ok with the 'dogs and smileys' being returned but:
- we're implying useage/policy with them and - whereas the goal is to get the moon phase; and i think we want to allow others to apply the rules for their wallet or app as to what it would mean, 
- I think that people will definitely want the ability to retrieve the moon phase alone without any implied illustration or hardcoded meaning. 

So I'm wondering if there could be an arg where it could return  U+1F311 through U+1F318? (🌑🌒🌓🌔🌕🌖🌗🌘)   with your dogs/smileys and suggested useage/actions as an alternate?

Such as:

`

char moon(int moonMode)
{
  int D,c,e;
  double jd;                      // Julian days

  time_t t = time(NULL);          // get time
  struct tm tm = *localtime(&t);  // get localtime
  int y = tm.tm_year + 1900;      // year, because its only above 1900, we add 1900 to the year
  int m = tm.tm_mon;              // month
  int d = tm.tm_mday - 2;         // day, because it can show the moon with 2 days in advance, we subtract

  if (m < 3) {
    y--;
    m += 12;
  }
  ++m;

  c = 365.25*y;                   // 365.25 total days in a year
  e = 30.6*m;
  jd = c+e+d-694039.09;           // Julian days
  jd /= 29.53;                    // moons cycle 29.53 days
  D = jd;	                        // convert decimal to int 	   
  jd -= D;                        // get negative of Julian days
  D = jd*8 + 1.5;	                // get fraction from 0-8 by rounding 0.5 to it               


  if (moonMode>0)
	{
  switch (D & 8) 
{                // because the new moon = 0 is the same on 8 we convert to 0
    case 1:
      return "🌚";               // New Moon Hype / Discounts
    case 2:
      return "🐶";               // Show Doge wen no moon
    case 3:
      return "🌓";               // First Quarter Moon
    case 4:
      return "🐶";               // Show Doge wen no moon
    case 5:
      return "🌝";               // Full Moon DownTrend / No Discounts
    case 6:
      return "🐶";               // Show Doge wen no moon
    case 7:
      return "🌗";               // Third Quarter
    case 8:
      return "🐶";               // Show Doge wen no moon
  }
  return 0;
}
else
{
  switch (D & 8) {                // because the new moon = 0 is the same on 8 we convert to 0
    case 1:
      return "🌑";               // New Moon 
    case 2:
      return "🌒";               // Waxing Crescent
    case 3:
      return "🌓";               // First Quarter Moon
    case 4:
      return "🌔";               // Waxing Gibbous
    case 5:
      return "🌕";               // Full Moon 
    case 6:
      return "🌖";               // Waning Gibbous
    case 7:
      return "🌗";               // Third Quarter
    case 8:
      return "🌘";               // Waning Crescent
  }
  return 0;
}

`

A final nit that makes me twitch is -- I hate single super abbreviated variables (D,c,e,jd,t,tm,y,m,d) but i think we can leave them for now if the calc works and just expand them later on for clarity.

Reply:
I reverted changes :)